File: Harvard - CS50/6pset/sentimental/vigenere/vigenere.py
from sys import argv
from cs50 import get_string

if (len(argv) == 2):

    key = argv[1]
    for letter in key:
        if not letter.isalpha():
            exit(1)

    word = get_string("plaintext: ")
    i = 0

    print("ciphertext: ", end="")
    for letter in word:

        if key[i].isupper():
            cipher = ord(key[i]) % 65
        else:
            cipher = ord(key[i]) % 97

        if(letter.islower()):
            print(chr(ord('a') + ((ord(letter) % 97) + cipher) % 26 ), end="")
            i += 1
        elif(letter.isupper()):
            print(chr(ord('A') + ((ord(letter) % 65) + cipher) % 26 ), end="")
            i += 1
        else:
            print(letter, end="")

        if i == len(key):
            i = 0
    print()
else:
    exit(1)

File: Harvard - CS50/2pset/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    if (argc != 2)
        return 1;

    string k = argv[1];
    int keylength = strlen(k);
    int putValue[keylength];
    for (int i =0; i < keylength; i++){
        if (islower(k[i]))
        {
            putValue[i] = (k[i] % 97);
        }
        else if (isupper(k[i]))
        {
            putValue[i] = (k[i] % 65);

        }
        else
        {
            return 1;
        }
    }
    string plaintext = get_string("plaintext: ");

    printf("ciphertext: ");
    for (int i = 0, j = 0, n = strlen(plaintext); i < n; i++)
    {
        if (islower(plaintext[i]))
        {
            int getCharPlace = plaintext[i] % 97;
            printf("%c", 'a' + ((getCharPlace + putValue[j%keylength] ) % 26));
            j++;
        }
        else if (isupper(plaintext[i]))
        {
            int getCharPlace = plaintext[i] % 65;
            printf("%c", 'A' + ((getCharPlace + putValue[j%keylength] ) % 26));
            j++;
        }
        else
        {
            printf("%c", plaintext[i]);
        }
    }
    printf("\n");
}